### PR TITLE
Get debug settings from user preference and send it to debuggee

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: src/configurationProvider.ts [hunk 1]
```diff
@@ -217,6 +223,57 @@ function configLogLevel(level) {
     return commands.executeJavaLanguageServerCommand(commands.JAVA_CONFIG_LOG_LEVEL, convertLogLevel(level));
 }
 
+interface IDebugSettings {
+    show_hex?: boolean;
```
Comment (testforstephen) [inline src/configurationProvider.ts]: Use `Camel` to keep name convention consistent.

File: src/configurationProvider.ts [hunk 2]
```diff
@@ -217,6 +223,57 @@ function configLogLevel(level) {
...
+        await updateDebugSettings();
+    } catch (err) {
+        // log a warning message and continue, since update settings failure should not block debug session
+        console.log("Cannot update debug settings.", err)
+    }
+}
+
+function collectDebugSettings(): IDebugSettings {
+    const settings: IDebugSettings = {};
+    if (vscode.workspace.getConfiguration().get("java.debug.settings.showHex")) {
```
Comment (testforstephen) [inline src/configurationProvider.ts]: Cannot set the value to false. no need to `if`.

File: package.json [hunk 3]
```diff
@@ -196,6 +196,22 @@
             "info",
             "verbose"
           ]
+        },
+        "java.debug.settings.showHex": {
+          "type": "boolean",
+          "default": false
```
Comment (testforstephen) [inline package.json]: add description here.

File: README.md [hunk 4]
```diff
@@ -69,6 +69,10 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
 ### User Settings
 
 - `java.debug.logLevel`: minimum level of debugger logs that are sent to VS Code, defaults to `warn`.
+- `java.debug.settings.showHex`: whether or not the number should be displayed as hex format in variable view, defaults to `false`.
```
Comment (akaroml) [inline README.md]: show integers in hex format

File: README.md [hunk 5]
```diff
@@ -69,6 +69,10 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+- `java.debug.settings.showStaticVariables`: whether or not the static variables will be displayed in variable view, defaults to `true`.
```
Comment (akaroml) [inline README.md]: show static variables in "Variables" viewlet

File: README.md [hunk 6]
```diff
@@ -69,6 +69,10 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+- `java.debug.settings.showQualifiedNames`: whether or not the class name should be displayed using the fully qualified class name, defaults to `false`.
```
Comment (akaroml) [inline README.md]: show fully qualified class names

File: package.json [hunk 7]
```diff
@@ -190,12 +190,33 @@
...
           "enum": [
             "error",
             "warn",
             "info",
             "verbose"
           ]
+        },
+        "java.debug.settings.showHex": {
+          "type": "boolean",
+          "description" : "whether or not the number should be displayed as hex format in variable view",
```
Comment (akaroml) [inline package.json]: Please change the descriptions accordingly.

Comment (akaroml) [reply package.json]: Same for below ones.

File: README.md [hunk 8]
```diff
@@ -69,6 +69,10 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
 ### User Settings
 
 - `java.debug.logLevel`: minimum level of debugger logs that are sent to VS Code, defaults to `warn`.
+- `java.debug.settings.showHex`: show the number in hex format in variable view, defaults to `false`.
```
Comment (akaroml) [inline README.md]: show numbers in hex format in "Variables" viewlet, de...

File: README.md [hunk 9]
```diff
@@ -69,6 +69,10 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
...
+- `java.debug.settings.showStaticVariables`: show the static variables in variable view, defaults to `true`.
+- `java.debug.settings.showQualifiedNames`: show the fully qualified class name in variable view, defaults to `false`.
```
Comment (akaroml) [inline README.md]: class names.

File: README.md [hunk 10]
```diff
@@ -69,6 +69,10 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
 ### User Settings
 
 - `java.debug.logLevel`: minimum level of debugger logs that are sent to VS Code, defaults to `warn`.
+- `java.debug.settings.showHex`: show numbers in hex format in "Variables" viewlet, defaults to `false`.
+- `java.debug.settings.showStaticVariables`: show static variables in "Variables" viewlet, defaults to `true`.
+- `java.debug.settings.showQualifiedNames`: show the fully qualified class names in "Variables" viewlet, defaults to `false`.
```
Comment (akaroml) [inline README.md]: remove "the" before fully....

**Pull request merged**